### PR TITLE
Creates a CSP PD Responses page, reporting on responses to CSP PD Unit 3.

### PR DESCRIPTION
**Pull request by ashercodeorg**
Further work on this page may happen after we receive more responses, including the insertion of aggregates.

### CODE REVIEW

File: dashboard/app/controllers/brook/brook_controller.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class BrookController < ApplicationController
+  def csppd3_responses
+    render local: {asher: 1}
```
Comment (Hamms) [inline dashboard/app/controllers/brook/brook_controller.rb]: I'm not sure what this (and the `csppd3_responses.html.haml` template) are supposed to represent. I'm assuming they're placeholders of some kind?

File: dashboard/app/controllers/reports_controller.rb [hunk 2]
```diff
@@ -57,14 +57,22 @@ def all_usage
   end
 
   def search_for_teachers
-    authorize! :read, :reports
```
Comment (laurelfan) [inline dashboard/app/controllers/reports_controller.rb:60]: why don't you need this line anymore?

File: dashboard/app/controllers/reports_controller.rb [hunk 3]
```diff
@@ -56,13 +56,14 @@ def all_usage
...
 
-    email_filter = "%#{params[:emailFilter]}%"
-    address_filter = "#{params[:addressFilter]}%"
-
-    @teachers = User.limit(100).where(user_type: 'teacher').where("email LIKE ?", email_filter).where("full_address LIKE ?", address_filter).joins(:followers).group('followers.user_id').pluck(:id, :name, :email, :full_address, 'COUNT(followers.id) AS num_students')
+    @headers = ['ID', 'Level ID', 'Data']
+    @responses3911 = LevelSource.limit(50).where(level_id: 3911).pluck(:id, :level_id, :data)
+    @responses3909 = LevelSource.limit(50).where(level_id: 3909).pluck(:id, :level_id, :data)
+    @responses3910 = LevelSource.limit(50).where(level_id: 3910).pluck(:id, :level_id, :data)
+    @responses3907 = LevelSource.limit(50).where(level_id: 3907).pluck(:id, :level_id, :data)
```
Comment (Hamms) [inline dashboard/app/controllers/reports_controller.rb]: You could instead do something like this:

``` ruby
@responses = {}
for level_id in [3911, ...]
  @responses[level_id] = LevelSource...
end
```

and then in the haml file, just iterate through `@responses`

File: dashboard/app/controllers/brook/brook_controller.rb (hunk 1 above)
Comment (ashercodeorg) [inline dashboard/app/controllers/brook/brook_controller.rb]: Removed, remnant garbage.

File: dashboard/config/routes.rb [hunk 4]
```diff
@@ -172,6 +172,9 @@ def redirect_to_teacher_dashboard
   post '/milestone/:user_id/level/:level_id', :to => 'activities#milestone', :as => 'milestone_level'
   post '/milestone/:user_id/:script_level_id', :to => 'activities#milestone', :as => 'milestone'
 
+  # one-off internal reports
+  get '/admin/brook/csppd', to: 'reports#csp_pd_responses', as: 'csp_pd_responses'
```
Comment (Hamms) [inline dashboard/config/routes.rb:176]: where is "brook" coming from?

**Comment by Hamms** [comment]
LGTM after comments

File: dashboard/config/routes.rb (hunk 4 above)
Comment (ashercodeorg) [inline dashboard/config/routes.rb:176]: The page is a one-off request from Brook.

File: dashboard/app/controllers/reports_controller.rb (hunk 3 above)
Comment (ashercodeorg) [inline dashboard/app/controllers/reports_controller.rb]: Done. Though I'm not sure the most cleanly (it feels like I'm using one more level of nesting in the HAML than I should be)...

**Comment by ashercodeorg** [comment]
Elijah: Please take another look when you get the chance.

File: dashboard/app/views/reports/csp_pd_responses.html.haml [hunk 5]
```diff
@@ -0,0 +1,24 @@
+%h1
+  CS Principles PD Responses
+
+- if @responses.size > 0
+  - @responses.each do |response|
```
Comment (Hamms) [inline dashboard/app/views/reports/csp_pd_responses.html.haml]: @responses is a hash, so wouldn't you want to do something like

``` ruby
- @responses.each do |level_id, level_sources|
  %h3
    Responses for Level
    = level_id
  %table
  - @headers ...
  - level_sources.each
```

Comment (ashercodeorg) [inline dashboard/app/views/reports/csp_pd_responses.html.haml]: Done.

**Comment by Hamms** [comment]
LGTM!

**Pull request merged**